src/megatron/bridge/training/config.py

```
@@ -1355,6 +1363,51 @@ def _validate_and_apply_deterministic_mode(self) -> None:
+        Raises:
+            AssertionError: If PyTorch version is < 2.3.0 or moe_token_dispatcher_type is 'flex'.
+            ValueError: If in-process restart is enabled.
+        """
+        if not self.dist.fake_process_group:
+            return
+
+        assert is_torch_min_version("2.3.0"), "Fake process group is only supported with PyTorch 2.3.0 and above."
+
+        # Check MoE token dispatcher type compatibility
+        if hasattr(self.model, "moe_token_dispatcher_type"):
+            assert self.model.moe_token_dispatcher_type != "flex", (
```

Ruff (0.14.14)

Possible hardcoded password assigned to: "moe_token_dispatcher_type"

(S105)